src/opencl_common.c

```diff
@@ -2829,6 +2881,273 @@ int opencl_calc_min_kpc(size_t lws, size_t gws, int v_width)
 	return gws * v_width;
 }
 
+static void buf_list_init(struct opencl_buffers **list) {
```

Comment:
Idem. To avoid duplication, I can use pseudo-generics. But, the source code becomes hard to follow.

Reply:
Very short and concise functions, probably better to keep them as-is for readability.